src/frontends/src/query_frontend.hpp

```diff
@@ -228,15 +170,19 @@ class RequestHandler {
       std::string json_content, std::string name,
       std::vector<VersionedModelId> models, std::string policy,
       long latency_slo_micros, InputType input_type) {
-    std::istringstream is(json_content);
-    ptree pt;
-    read_json(is, pt);
+    rapidjson::Document d;
```

Comment:
Actually, this should be moved into `json_util.hpp`. Can do that in a future commit but the idea (for error handling and the output error message would be the same). `rapidjson::GetParseError_En(ok.Code())` outputs a useful error message as well as an offset `ok.Offset()` in the JSON string, but does not get more detailed than that.